docs/devguide.rst

```diff
@@ -61,22 +61,66 @@ is open to interpretation.
   unittest directly or write tests in an object-oriented style.
 * Test generators make more dots and the dots must flow!
 
+You can easily check for style issues, including some outright bugs such
+as mispelled variable names, using pylint. If you're using Anaconda you'll
+need to run "conda install pylint" once. You can easily run pylint on
+the edited files in your uncommited git change:
+
+    $ make lint
+
+If you want to lint the entire code base run:
+
+    $ make lint-all
+
 How to Test
 ================
-First, install nose: http://nose.readthedocs.org/en/latest/
+First, install nose: http://nose.readthedocs.org/en/latest/. Second, ensure
+your cwd is the root directory of the project (i.e., the one containing the
+.git directory).
 
 To perform all unit tests::
 
-    $ cd tests/
-    $ nosetests
+    $ make test-all
```

Comment:
> Using make test-all should be an alternative, not the solely recommended way. 

That's why I list three ways to do it. If someone wants to invoke nosetests directly I show how to do so in order to avoid picking up local dependencies. And that includes depending on the parser_tables.py module that is somewhere in your PYTHONPATH from having installed xonsh. If you run the tests by typing

    cd tests/
    nosetests

you're not actually testing the code in the ../xonsh directory; assuming you haven't explicitly included that directory at the front of PYTHONPATH, of course. You can see this by running "ls -lutr xonsh" and noticing that the access timestamps haven't changed. If you do the same ls command on the directory where you installed xonsh you'll see that those modules are accessed when you run the tests using the original directions.